### PR TITLE
Modernize various files (src/m*)

### PR DESCRIPTION
**Pull request by nerzhul**
* range-based for loops
* code style
* C++ headers instead of C headers
* Default operators
* empty function

Thanks to clang-tidy

### CODE REVIEW

File: src/map.cpp [hunk 1]
```diff
@@ -527,14 +527,13 @@ void Map::unloadUnreferencedBlocks(std::vector<v3s16> *unloaded_blocks)
...
-		j != sectorList.end(); ++j) {
-		MapSector *sector = m_sectors[*j];
+	for (v2s16 j : sectorList) {
+		MapSector *sector = m_sectors[j];
 		// If sector is in sector cache, remove it from there
 		if(m_sector_cache == sector)
 			m_sector_cache = NULL;
 		// Remove from map and delete
-		m_sectors.erase(*j);
+		m_sectors.erase(j);
```
Comment (SmallJoker) [inline src/map.cpp]: Range-based for loops aren't made for modifying the container. I'd rather like to see the regular loop here.

File: src/map.h [hunk 2]
```diff
@@ -107,11 +107,7 @@ struct MapEditEvent
 		case MEET_OTHER:
 		{
 			VoxelArea a;
-			for(std::set<v3s16>::iterator
-					i = modified_blocks.begin();
-					i != modified_blocks.end(); ++i)
-			{
-				v3s16 p = *i;
+			for (auto p : modified_blocks) {
```
Comment (SmallJoker) [inline src/map.h]: `v3s16 &p : `

File: src/modalMenu.h [hunk 3]
```diff
@@ -111,16 +111,12 @@ class GUIModalMenu : public gui::IGUIElement
 	{
 		const core::list<gui::IGUIElement*> &children = getChildren();
 		core::list<gui::IGUIElement*> children_copy;
-		for(core::list<gui::IGUIElement*>::ConstIterator
-				i = children.begin(); i != children.end(); i++)
-		{
-			children_copy.push_back(*i);
+		for (gui::IGUIElement* i : children) {
```
Comment (SmallJoker) [inline src/modalMenu.h]: `gui::IGUIElement *i`

File: src/modalMenu.h [hunk 4]
```diff
@@ -111,16 +111,12 @@ class GUIModalMenu : public gui::IGUIElement
...
+			children_copy.push_back(i);
 		}
-		for(core::list<gui::IGUIElement*>::Iterator
-				i = children_copy.begin();
-				i != children_copy.end(); i++)
-		{
-			(*i)->remove();
+		
+		for (gui::IGUIElement* i : children_copy) {
```
Comment (SmallJoker) [inline src/modalMenu.h]: `gui::IGUIElement *i`

File: src/mods.cpp [hunk 5]
```diff
@@ -131,23 +128,16 @@ std::vector<ModSpec> flattenMods(std::map<std::string, ModSpec> mods)
...
+ModConfiguration::ModConfiguration(const std::string &worldpath)
 {
 }
 
 void ModConfiguration::printUnsatisfiedModsError() const
 {
-	for (std::vector<ModSpec>::const_iterator it = m_unsatisfied_mods.begin();
-		it != m_unsatisfied_mods.end(); ++it) {
-		ModSpec mod = *it;
+	for (const auto &mod : m_unsatisfied_mods) {
```
Comment (SmallJoker) [inline src/mods.cpp]: `const ModSpec &mod`

File: src/mods.cpp [hunk 6]
```diff
@@ -131,23 +128,16 @@ std::vector<ModSpec> flattenMods(std::map<std::string, ModSpec> mods)
...
 		errorstream << "mod \"" << mod.name << "\" has unsatisfied dependencies: ";
-		for (std::unordered_set<std::string>::iterator dep_it =
-			mod.unsatisfied_depends.begin();
-			dep_it != mod.unsatisfied_depends.end(); ++dep_it)
-			errorstream << " \"" << *dep_it << "\"";
+		for (const auto &unsatisfied_depend : mod.unsatisfied_depends)
```
Comment (SmallJoker) [inline src/mods.cpp]: `const std::string &unsatisfied_depend`

File: src/mods.cpp [hunk 7]
```diff
@@ -175,17 +165,15 @@ void ModConfiguration::addMods(const std::vector<ModSpec> &new_mods)
 
 		std::set<std::string> seen_this_iteration;
 
-		for (std::vector<ModSpec>::const_iterator it = new_mods.begin();
-				it != new_mods.end(); ++it) {
-			const ModSpec &mod = *it;
-			if(mod.part_of_modpack != (bool)want_from_modpack)
+		for (const auto &mod : new_mods) {
```
Comment (SmallJoker) [inline src/mods.cpp]: `const ModSpec &mod`

File: src/map.cpp (hunk 1 above)
Comment (nerzhul) [inline src/map.cpp]: i need to verify the norm but if language permit it i'm pretty sure it's good :)

**Comment by nerzhul** [comment]
thanks for the review, i will fix the auto points later

File: src/mapgen_v6.cpp [hunk 8]
```diff
@@ -847,21 +849,20 @@ void MapgenV6::flowMud(s16 &mudflow_minpos, s16 &mudflow_maxpos)
 				}
 
 				// Drop mud on side
-				for (u32 di = 0; di < 4; di++) {
-					v3s16 dirp = dirs4[di];
+				for (const v3s16 &dirp : dirs4) {
```
Comment (paramat) [inline src/mapgen_v6.cpp]: Will this still iterate through the directions in the same order as before?

Comment (nerzhul) [inline src/mapgen_v6.cpp]: yes, for range-based loop iterate in incremental order, if you want other order you cannot use it

File: src/mg_ore.cpp [hunk 9]
```diff
@@ -64,8 +64,8 @@ size_t OreManager::placeAllOres(Mapgen *mg, u32 blockseed,
 
 void OreManager::clear()
 {
-	for (size_t i = 0; i < m_objects.size(); i++) {
-		Ore *ore = (Ore *)m_objects[i];
+	for (auto &object : m_objects) {
```
Comment (paramat) [inline src/mg_ore.cpp]: Same order?

**Comment by paramat** [comment]
Mapgen stuff looks good, as long as the new for loops keep the same order.

File: src/mapgen_v6.cpp (hunk 8 above)
Comment (paramat) [inline src/mapgen_v6.cpp]: Ok.

File: src/mg_ore.cpp (hunk 9 above)
Comment (nerzhul) [inline src/mg_ore.cpp]: i answer to you just on previous comment, look at the norm examples here: http://en.cppreference.com/w/cpp/language/range-for

Comment (paramat) [inline src/mg_ore.cpp]: I guess so.

Comment (paramat) [inline src/mg_ore.cpp]: Fine, that explains 'auto'.

Comment (nerzhul) [inline src/mg_ore.cpp]: auto should be used essential for iterators to reduce litterature of the iterator name, but here i should replace it

File: src/map.cpp (hunk 1 above)
Comment (nerzhul) [inline src/map.cpp]: @SmallJoker you  (and me) misread the block :) we are iterating on `for (v2s16 j : sectorList)` not m_sectors :)

**Pull request merged**